launch/launch/actions/execute_process.py

```diff
@@ -84,7 +84,98 @@
 
 @expose_action('executable')
 class ExecuteProcess(Action):
-    """Action that begins executing a process and sets up event handlers for the process."""
+    """
+    Action that begins executing a process and sets up event handlers for it.
+
+    Simple example:
+
+        .. doctest::
+
+            >>> ld = LaunchDescription([
+            ...     ExecuteProcess(
+            ...         cmd=['ls', '-las'],
+            ...         name='my_ls_process',  # this is optional
+            ...         output='both',
+            ...     ),
+            ... ])
+
+        .. code-block:: xml
+
+            <launch>
+                <executable cmd="ls -las" name="my_ls_process" output="both"/>
+            </launch>
+
+    Substitutions in the command:
+
+        .. doctest::
+
+            >>> ld = LaunchDescription([
+            ...     DeclareLaunchArgument(name='file_path', description='file path to cat'),
+            ...     ExecuteProcess(
+            ...         # each item of the command arguments' list can be:
+            ...         # a string ('cat'),
+            ...         # a substitution (`LaunchConfiguration('file_path')`),
+            ...         # or a list of string/substitutions
+            ...         # (`[LaunchConfiguration('directory'), '/file.txt']`)
+            ...         cmd=['cat', LaunchConfiguration('file_path')],
+            ...     ),
+            ... ])
+
+        .. code-block:: xml
+
+            <launch>
+                <arg name="file_path" description="path of the file to cat"/>
+                <executable cmd="cat $(var file_path)"/>
+            </launch>
+
+    Optional cli argument:
+
+        .. doctest::
+
+            >>> ld = LaunchDescription([
+            ...     DeclareLaunchArgument(name='open_gui', default_value='False'),
+            ...     ExecuteProcess(
+            ...         cmd=['my_cmd', '--open-gui'],
+            ...         condition=IfCondition(LaunchConfiguration('open_gui')),
+            ...     ),
+            ...     ExecuteProcess(
+            ...         cmd=['my_cmd'],
+            ...         condition=UnlessCondition(LaunchConfiguration('open_gui')),
+            ...     ),
+            ... ])
+
+        .. code-block:: xml
+
+            <launch>
+                <arg name="open_gui" description="when truthy, the gui will be opened"/>
+                <executable cmd="my_cmd --open-gui" if="$(var open_gui)"/>
+                <executable cmd="my_cmd" unless="$(var open_gui)"/>
+                <!--If you have a launch configuration named 'open_gui_arg' that is equal
+                    to "--open-gui" when the "open_gui" argument is truthy and "" when it's
+                    falsy, you'd think that avoiding conditions is possible and you can only
+                    use one executable action with `cmd="my_cmd $(var open_gui_arg)"`.
+                    Unfortunately, that doesn't work, use conditions as in this example.-->
```

Comment:
Hmm, why not? This is unusual.

Reply:
That ends up being an empty argument passed to the process, depending on the argument parser that might be an issue.
e.g. I think that `rcl` has problems with that (if it appears after `--ros-args`).

Reply:
I see. It should be straightforward to filter them before process invocation though (not to do that now, only pointing out we could easily fix that).

Reply:
okay, I will expand the comment explaining the difference between the two options.

Reply:
I deleted the comment, as I think it was confusing.
PTAL